scripts/bgpv4_forwarding.go

```diff
@@ -0,0 +1,216 @@
+			log.Println("WARNING:", w)
+		}
+	default:
+		log.Fatal("Unknown response type:", v)
+	}
+}
+
+func newConfig() (gosnappi.GosnappiApi, gosnappi.Config) {
+	// create a new API handle to make API calls against otgHost
+	api := gosnappi.NewApi()
+	api.NewHttpTransport().SetLocation(otgHost).SetVerify(false)
+
+	// create an empty traffic configuration
+	config := api.NewConfig()
+	// create traffic endpoints
+	p1 := config.Ports().Add().SetName("p1").SetLocation(otgPort1)
+	p2 := config.Ports().Add().SetName("p2").SetLocation(otgPort2)
+
+	// create devices
+	p1d1 := config.Devices().Add().SetName("p1d1")
+	p1d1Eth := p1d1.Ethernets().Add().
+		SetName("p1d1eth").
+		SetPortName(p1.Name()).
+		SetMac(srcMAC)
+	p1d1IPv4 := p1d1Eth.Ipv4Addresses().Add().
+		SetName("p1d1Ipv4").
+		SetAddress(srcIP).
+		SetGateway(srcGateway).
+		SetPrefix(int32(srcPrefix))
+
+	p1d1BGP := p1d1.Bgp().
+		SetRouterId(p1d1IPv4.Address())
+	p1d1BGPv4Peer := p1d1BGP.Ipv4Interfaces().Add().
+		SetIpv4Name(p1d1IPv4.Name()).
+		Peers().Add().
+		SetName("p1d1BGPv4Peer").
+		SetPeerAddress(p1d1IPv4.Gateway()).
+		SetAsNumber(int32(srcAS)).
+		SetAsType(gosnappi.BgpV4PeerAsType.EBGP)
+
+	p1d1BGPv4PeerRoutes := p1d1BGPv4Peer.V4Routes().Add().
+		SetName("p1d1BGPv4PeerR1").
+		SetNextHopIpv4Address(p1d1IPv4.Address()).
+		SetNextHopAddressType(gosnappi.BgpV4RouteRangeNextHopAddressType.IPV4).
+		SetNextHopMode(gosnappi.BgpV4RouteRangeNextHopMode.MANUAL)
+	p1d1BGPv4PeerRoutes.Addresses().Add().
+		SetAddress("10.10.10.1").
+		SetPrefix(24).
+		SetCount(5).
+		SetStep(1)
+
+	p2d1 := config.Devices().Add().SetName("p2d1")
+	p2d1Eth := p2d1.Ethernets().Add().
+		SetName("p2d1eth").
+		SetPortName(p2.Name()).
+		SetMac(dstMAC)
+	p2d1IPv4 := p2d1Eth.Ipv4Addresses().Add().
+		SetName("p2d1Ipv4").
+		SetAddress(dstIP).
+		SetGateway(dstGateway).
+		SetPrefix(int32(dstPrefix))
+
+	p2d1BGP := p2d1.Bgp().
+		SetRouterId(p2d1IPv4.Address())
+	p2d1BGPv4Peer := p2d1BGP.Ipv4Interfaces().Add().
+		SetIpv4Name(p2d1IPv4.Name()).
+		Peers().Add().
+		SetName("p2d1BGPv4Peer").
+		SetPeerAddress(p2d1IPv4.Gateway()).
+		SetAsNumber(int32(dstAS)).
+		SetAsType(gosnappi.BgpV4PeerAsType.EBGP)
+
+	p2d1BGPv4PeerRoutes := p2d1BGPv4Peer.V4Routes().Add().
+		SetName("p2d1BGPv4PeerR1").
+		SetNextHopIpv4Address(p2d1IPv4.Address()).
+		SetNextHopAddressType(gosnappi.BgpV4RouteRangeNextHopAddressType.IPV4).
+		SetNextHopMode(gosnappi.BgpV4RouteRangeNextHopMode.MANUAL)
+	p2d1BGPv4PeerRoutes.Addresses().Add().
+		SetAddress("20.20.20.1").
+		SetPrefix(24).
+		SetCount(5).
+		SetStep(1)
+
+	// create a flow and set the endpoints
+	f1 := config.Flows().Add().SetName("p1.v4.p2")
+
+	f1.TxRx().Device().SetTxNames([]string{p1d1BGPv4PeerRoutes.Name()}).SetRxNames([]string{p2d1BGPv4PeerRoutes.Name()})
+
+	// enable per flow metrics tracking
+	f1.Metrics().SetEnable(true)
+	// set size, count and transmit rate for all packets in the flow
+	f1.Size().SetFixed(512)
+	f1.Rate().SetPps(500)
```

Comment:
Maybe lower pps to 100 or so for lower end setups.

Reply:
Done